network-primitives/full_node_provisioner.tf

```diff
@@ -108,26 +107,46 @@ resource "null_resource" "start-full-nodes" {
     destination = "/subspace/bootstrap_node_keys.txt"
   }
 
-  # copy compose file
+  # copy compose file creation script
   provisioner "file" {
     source      = "${var.path-to-scripts}/create_full_node_compose_file.sh"
     destination = "/subspace/create_compose_file.sh"
   }
 
+  # copy subspace entry script
+  provisioner "file" {
+    source      = "${var.path-to-scripts}/subspace.sh"
+    destination = "/usr/bin/subspace"
+  }
+
+  # copy subspace systemd file
+  provisioner "file" {
+    source      = "${var.path-to-scripts}/subspace.service"
+    destination = "/etc/systemd/system/subspace.service"
+  }
+
   # start docker containers
   provisioner "remote-exec" {
     inline = [
-      "docker compose -f /subspace/docker-compose.yml down",
+      # stop any running service
+      "systemctl stop subspace.service",
+      "systemctl reenable subspace.service",
+      "systemctl daemon-reload",
+
+      # create .env file
       "echo NODE_ORG=${var.full-node-config.docker-org} > /subspace/.env",
       "echo NODE_TAG=${var.full-node-config.docker-tag} >> /subspace/.env",
       "echo NETWORK_NAME=${var.network-name} >> /subspace/.env",
       "echo NODE_ID=${count.index} >> /subspace/.env",
       "echo NODE_KEY=$(sed -nr 's/NODE_${count.index}_KEY=//p' /subspace/node_keys.txt) >> /subspace/.env",
+
+      # create docker compose file
       "sudo chmod +x /subspace/create_compose_file.sh",
+      "sudo chmod +x /usr/bin/subspace",
       "sudo /subspace/create_compose_file.sh ${var.bootstrap-node-config.reserved-only} ${length(local.full_node_ip_v4)} ${count.index} ${length(local.bootstrap_nodes_ip_v4)}",
-      "docker compose -f /subspace/docker-compose.yml up -d --remove-orphans",
-      "sudo mkdir -p /etc/iptables/",
-      "sudo iptables-save > /etc/iptables/rules.v4",
+
+      # start subspace node
+      "systemctl start subspace.service",
```

Comment:
I assume you've deleted `/etc/iptables` on those machines. Just checking.

Reply:
yes we did delete them.